core/src/serial/codecs/lines.rs

```diff
@@ -67,6 +72,34 @@ impl Decoder for LinesCodec {
             // Since our job is to find the delimiter, we don't need to re-read the bytes we have already looked at.
             self.cursor = read_to;
 
+            /* Handle disconnection */
+            if look_at.is_empty() && read_to == 0 && self.cursor == 0 && src.len() == 0 {
+                // This by itself is not enough to detect a disconnect, but it is a good indicator.
+                // We need to try to connect to the port to be sure that the device has gone away.
+                // If we get an 'busy' error, we are still connected.
+
+                // Try to connect to the port
+                let port = serialport::new(&self.serial_port_path, 9600)
```

Comment:
Hopefully there is a cleaner way to ask if a port is busy or not.
Please have a look.

Reply:
This is actually a bit tricky. I haven't found a better way. In the API, the closest thing is `available_ports`, which would list the ports that aren't busy and available, but we can't use that since devices that aren't connected wouldn't be shown either :(